File: src/Helpers/PromptDisplayHelpers.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;

/// <summary>
/// Provides methods for displaying prompt information.
/// </summary>
public static class PromptDisplayHelpers
{
    /// <summary>
    /// Gets a display name for the prompt directory location in the specified scope.
    /// </summary>
    /// <param name="scope">The scope to get the location display name for</param>
    /// <returns>A string representing the location, or null if the scope is invalid</returns>
    public static string? GetLocationDisplayName(ConfigFileScope scope)
    {
        var promptPath = PromptFileHelpers.FindPromptDirectoryInScope(scope);
        if (promptPath != null)
        {
            return $"{promptPath} ({scope.ToString().ToLower()})";
        }
        
        var scopeDir = ConfigFileHelpers.GetScopeDirectoryPath(scope);
        if (scopeDir == null) return null;

        return $"{Path.Combine(scopeDir, "prompts")} ({scope.ToString().ToLower()})";
    }

    /// <summary>
    /// Displays prompts for a specific scope.
    /// </summary>
    /// <param name="scope">The scope to display prompts for</param>
    public static void DisplayPrompts(ConfigFileScope scope)
    {
        // Get the location display name, which works even if the directory doesn't exist yet
        var locationDisplay = GetLocationDisplayName(scope);
        if (locationDisplay == null) return;
        
        ConsoleHelpers.WriteLine($"LOCATION: {locationDisplay}");
        Console.WriteLine();

        // Find the directory to check for prompts
        var promptDir = PromptFileHelpers.FindPromptDirectoryInScope(scope);
        
        // Check for prompts only if the directory exists
        var promptFiles = new List<string>();
        if (promptDir != null && Directory.Exists(promptDir))
        {
            promptFiles = Directory.GetFiles(promptDir, "*.prompt")
                .OrderBy(file => Path.GetFileNameWithoutExtension(file))
                .ToList();
        }

        DisplayPromptFiles(promptFiles);
    }

    /// <summary>
    /// Displays a list of prompt files.
    /// </summary>
    /// <param name="promptFiles">The list of prompt files to display</param>
    /// <param name="indentLevel">The level of indentation for display</param>
    public static void DisplayPromptFiles(List<string> promptFiles, int indentLevel = 2)
    {
        var indent = new string(' ', indentLevel);
        
        if (promptFiles.Count == 0)
        {
            ConsoleHelpers.WriteLine($"{indent}No prompts found in this scope");
            return;
        }

        foreach (var promptFile in promptFiles)
        {
            var promptName = Path.GetFileNameWithoutExtension(promptFile);
            ConsoleHelpers.WriteLine($"{indent}/{promptName}");
        }
    }

    /// <summary>
    /// Displays information about saved prompt files.
    /// </summary>
    /// <param name="filesSaved">List of saved files</param>
    public static void DisplaySavedPromptFiles(List<string> filesSaved)
    {
        var firstFileSaved = filesSaved.First();
        var additionalFiles = filesSaved.Skip(1).ToList();

        ConsoleHelpers.WriteLine($"Saved: {firstFileSaved}\n");

        var hasAdditionalFiles = additionalFiles.Any();
        if (hasAdditionalFiles)
        {
            foreach (var additionalFile in additionalFiles)
            {
                ConsoleHelpers.WriteLine($"  and: {additionalFile}");
            }
         
            ConsoleHelpers.WriteLine();
        }

        var promptName = Path.GetFileNameWithoutExtension(firstFileSaved);
        ConsoleHelpers.WriteLine($"USAGE: /{promptName} (in chat)");
    }
}

File: ts/src/Helpers/PromptDisplayHelpers.ts
import { ConfigFileScope } from '../Configuration/ConfigFileScope';
import { ConsoleHelpers } from './ConsoleHelpers';
import { PromptFileHelpers } from './PromptFileHelpers';
import { PathHelpers } from './PathHelpers';

/**
 * Provides methods for displaying prompt information.
 */
export class PromptDisplayHelpers {
  
  /**
   * Gets a display name for the prompt directory location in the specified scope.
   */
  static getLocationDisplayName(scope: ConfigFileScope): string | null {
    const promptPath = PromptFileHelpers.findPromptDirectoryInScope(scope);
    if (promptPath) {
      return this.formatLocationHeader(promptPath, scope);
    }
    
    const scopeDir = PathHelpers.getScopeDirectoryPath(scope);
    if (!scopeDir) return null;

    const promptsDir = require('path').join(scopeDir, 'prompts');
    return this.formatLocationHeader(promptsDir, scope);
  }

  /**
   * Formats a location header for display.
   */
  static formatLocationHeader(path: string, scope: ConfigFileScope): string {
    return `${path} (${scope.toString().toLowerCase()})`;
  }

  /**
   * Displays prompts for a specific scope.
   */
  static displayPrompts(scope: ConfigFileScope): void {
    const locationDisplay = this.getLocationDisplayName(scope);
    if (!locationDisplay) return;
    
    const promptNames = PromptFileHelpers.listPromptsInScope(scope);
    this.displayItemList(locationDisplay, promptNames);
  }

  /**
   * Displays a list of items with a location header.
   */
  static displayItemList(locationHeader: string, items: string[]): void {
    ConsoleHelpers.writeLine(`LOCATION: ${locationHeader}`);
    ConsoleHelpers.writeLine('');

    if (items.length === 0) {
      ConsoleHelpers.writeLine('  No items found in this scope');
    } else {
      for (const item of items) {
        ConsoleHelpers.writeLine(`  ${item}`);
      }
    }
  }

  /**
   * Displays a single prompt with content preview.
   */
  static displayPrompt(name: string, promptFilePath: string, scope: ConfigFileScope, content: string): void {
    const location = this.formatLocationHeader(promptFilePath, scope);
    this.displayItem(name, content, location, true);
  }

  /**
   * Displays a single item with content.
   */
  static displayItem(name: string, content: string, location: string, _limitValue: boolean = false): void {
    ConsoleHelpers.writeLine(`LOCATION: ${location}`);
    ConsoleHelpers.writeLine('');
    ConsoleHelpers.writeLine(`  ${name}`);
    ConsoleHelpers.writeLine('');
    
    // Display content with indentation
    const lines = content.split('\n');
    for (const line of lines) {
      ConsoleHelpers.writeLine(`    ${line}`);
    }
  }

  /**
   * Displays information about saved prompt files.
   */
  static displaySavedPromptFile(fileName: string): void {
    ConsoleHelpers.writeLine(`Saved: ${fileName}`);
    ConsoleHelpers.writeLine('');
    
    // Extract prompt name from filename
    const path = require('path');
    const promptName = path.basename(fileName, '.prompt');
    ConsoleHelpers.writeLine(`USAGE: /${promptName} (in chat)`);
  }
}